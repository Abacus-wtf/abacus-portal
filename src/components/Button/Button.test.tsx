import { render, fireEvent, getByText } from "@test-utils/index"
import React from "react"
import "@testing-library/jest-dom/extend-expect"
import Button, { ButtonsWhite, ButtonClear } from "."

const defaultProps = {
  onClick: () => {
    //
  },
}

describe("Button", () => {
  it("Matches snapshot", () => {
    const component = render(<Button {...defaultProps} />)
    expect(component.container).toMatchSnapshot()
  })

  it("calls onClick when clicked", () => {
    const BUTTON_TEXT = "SUBMIT"
    const onClick = jest.fn()
    const { container } = render(
      <Button {...defaultProps} onClick={onClick}>
        {BUTTON_TEXT}
      </Button>
    )
    fireEvent(
      getByText(container, BUTTON_TEXT),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      })
    )
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})

describe("ButtonsWhite", () => {
  it("Matches snapshot", () => {
    const component = render(<ButtonsWhite {...defaultProps} />)
    expect(component.container).toMatchSnapshot()
  })

  it("calls onClick when clicked", () => {
    const BUTTON_TEXT = "SUBMIT"
    const onClick = jest.fn()
    const { container } = render(
      <ButtonsWhite {...defaultProps} onClick={onClick}>
        {BUTTON_TEXT}
      </ButtonsWhite>
    )
    fireEvent(
      getByText(container, BUTTON_TEXT),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      })
    )
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})

describe("ButtonClear", () => {
  it("Matches snapshot", () => {
    const component = render(<ButtonClear {...defaultProps} />)
    expect(component.container).toMatchSnapshot()
  })

  it("calls onClick when clicked", () => {
    const BUTTON_TEXT = "SUBMIT"
    const onClick = jest.fn()
    const { container } = render(
      <ButtonClear {...defaultProps} onClick={onClick}>
        {BUTTON_TEXT}
      </ButtonClear>
    )
    fireEvent(
      getByText(container, BUTTON_TEXT),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true,
      })
    )
    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
