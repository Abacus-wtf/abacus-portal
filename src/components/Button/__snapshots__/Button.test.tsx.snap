// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Matches snapshot 1`] = `
<div>
  <button
    class="sc-bdfBQB jjXNlU btn btn-primary"
    type="button"
  />
</div>
`;

exports[`ButtonClear Matches snapshot 1`] = `
<div>
  <button
    class="sc-bdfBQB sc-dlfnuX jjXNlU ipwiFM btn btn-primary"
    type="button"
  />
</div>
`;

exports[`ButtonsWhite Matches snapshot 1`] = `
<div>
  <button
    class="sc-bdfBQB sc-gsTEea jjXNlU dYSOdb btn btn-primary"
    type="button"
  />
</div>
`;
